### ISSUE TITLE
Migrate to the latest package_config

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/package_resolver#29]
This is needed to unblock package:test users from getting the latest package_config.

**Comment by jakemac53** [comment]
I will start on this and push it to a `package-config-2` branch which we can depend on from other things.

**Comment by jakemac53** [comment]
This definitely gets interesting for this package.... it exposes a lot of simple `Map<String, Uri>` as well as its own notion of "package root" which is now an explicit thing. We likely want to consider breaking changes :/.

**Comment by jakemac53** [comment]
It also looks like almost the entire value proposition of this package is no longer applicable - package_config 2.x adds most of this functionality

**Comment by jakemac53** [comment]
See https://github.com/dart-lang/package_resolver/issues/30 - we will likely close this in favor of deprecation

**Comment by jakemac53** [comment]
cc @lrhn do you think there is any value in this package any more?

**Comment by lrhn** [comment]
Probably not, but I don't know where this package is is used, or for what.

It seems like an abstraction layer over package resolution, which only makes sense if you are not using the platform's own package resolution (`Isolate.resolvePackageUri`). That might be useful for meta-programming.

Since we have removed the `packages` directory, that removes at least half of the reason to have this package. We have then introduced a different package configuration file, but this package doesn't recognize that yet, and doesn't provide all the features.
